### PR TITLE
theme: Make dark_plus.toml more accurate to VSCode

### PR DESCRIPTION
**Pull request by saccarosium**
The current implementation of the Dark+ has some inconsistencies with the original. I've made some modification, witch the most importants are:
1. Statusbar is now gray instead of blue, to make it more readable;
2. Whitespace are a more subtle color;
3. Indent Character is now distinguishable from the whitespaces.

## C

| Current | Updated | VSCode |
| --- | --- | --- |
| Screenshot 2023-11-04 at 21 30 00 | Screenshot 2023-11-04 at 21 28 33 | Screenshot 2023-11-04 at 21 29 43 |

## CPP

| Current | Updated | VSCode |
| --- | --- | --- |
| Screenshot 2023-11-04 at 21 41 31 | Screenshot 2023-11-04 at 21 41 19 | Screenshot 2023-11-04 at 21 41 14 |

### CODE REVIEW

**Comment by kirawi** [comment]
If I'm reading the changes correctly, it looks like the colors are actually inaccurate to the original theme. All the colors were taken directly from the definition. I'm also against changing the statusline color since it the theme is supposed to be a faithful port of the original.

**Comment by saccarosium** [comment]
Hi @kirawi,
You are right! Actually the diffs mismatch slide throw my radar. It is probably because I've crossed referenced also two vim colorschemes: [codedark.vim](https://github.com/tomasiser/vim-code-dark) and [vscode.nvim](https://github.com/Mofiqul/vscode.nvim). 

As for VSCode's source I checked this two: [dark+](https://github.com/microsoft/vscode/blob/main/extensions/theme-defaults/themes/dark_plus.json) and [vsdark](https://github.com/microsoft/vscode/blob/main/extensions/theme-defaults/themes/dark_vs.json)

>  I'm also against changing the statusline color since it the theme is supposed to be a faithful port of the original.

I think the statusline's color can be discussed, some people could prefer that.

**Comment by kirawi** [comment]
Neither contain the full definition of the theme. What I had to do was open up the command palette and run `Generate color theme from current settings`.

File: runtime/themes/dark_plus.toml [hunk 1]
```diff
@@ -1,138 +1,107 @@
...
+DiffGreen = '#4B5632'
+DiffLightBlue = '#87d7ff'
+DiffBlue = '#005f87'
+SearchCurrent = '#515c6a'
+Search = '#613315'
+PopupFront = '#BBBBBB'
+PopupBack  = '#2D2D30'
+Selection = '#264F78'
+Cursor = '#A6A6A6'
+WhiteSpace = '#4D4D4D'
```
Comment (kirawi) [inline runtime/themes/dark_plus.toml:103]: Should be `Whitespace` set to `3e3e3d`

**Comment by kirawi** [comment]
Inlay hint colors were changed to the colors of Dark+ experimental in https://github.com/helix-editor/helix/pull/7611, but they are now defined as
```json
"editorInlayHint.background": "#6161611a",
"editorInlayHint.foreground": "#969696",
```

File: runtime/themes/dark_plus.toml (hunk 1 above)
Comment (kirawi) [reply runtime/themes/dark_plus.toml:103]: Actually, I prefer to keep the names in terms of colors. That was originally my intention with this theme, but I never found the time to finish that. Trust me, its less of a headache trying to find a semantic name for things.

**Comment by kirawi** [comment]
Actually, I see that they changed many colors in the theme. So the previous values will have to be rechecked anyway. The statusline change is okay then since they changed it.

**Comment by kirawi** [comment]
I'll get back to you once I've checked everything.

**Comment by saccarosium** [comment]
> What I had to do was open up the command palette and run Generate color theme from current settings

Ok, when I get home I will check it myself also.

For the change a think we stick to the original `dark+` theme instead of the new `dark modern`.

**Comment by saccarosium** [comment]
I've came back to the original structure and only made some modifications.

**Comment by kirawi** [comment]
Sorry, I didn't mean to indicate that all of your changes were unwanted. I liked how you previously formatted it, but it's fine if you don't want to do that work again. I agree that VSCode Dark Modern should be a separate theme and you are free to open up a new PR with the original changes for that. You would have more creative liberty that route since I only maintain Dark+ :P

**Comment by saccarosium** [comment]
> Sorry, I didn't mean to indicate that all of your changes were unwanted.

No problem I did realize a lot of the colors I was used to were influenced by the vim colorschemes I previously used. So there was a lot of personal taste I decided to rollback to your version. After all you are the maintainer.

> I liked how you previously formatted it, but it's fine if you don't want to do that work again.

Do you mean separated the names with comments and aligned everything?

**Comment by kirawi** [comment]
> So there was a lot of personal taste I decided to rollback to your version.

If it is close enough to the Vim version, then we could probably offer a Vim variant of Dark+. We already offer a Zed variant of onelight.

> Do you mean separated the names with comments and aligned everything?

Yeah, it looked nice.

**Comment by saccarosium** [comment]
> Did the formatting. Yeah, it looked nice.

**Comment by kirawi** [comment]
I think it would be acceptable, but I'll leave that to the discretion of the maintainers.

**Comment by David-Else** [comment]
Please note line 10 and 11, usize has now gone blue, is this a mistake? Please don't make any changes to the inlay hint colours or statusbar, thanks.

## This PR
Screenshot from 2023-11-06 22-02-43
## How it was
Screenshot from 2023-11-06 21-58-50

**Comment by saccarosium** [comment]
> Please note line 10 and 11, usize has now gone blue, is this a mistake?

In VSCode are green but it is inconsistent because it uses a regex highlight system. Since we are using TreeSitter, it does what we tell it to do, I've changed how it displays `built-in` types in blue and the rest of the types in green, witch does the right thing for C and C++.

**Comment by kirawi** [comment]
So, I think reverting that change in highlighting would be the more correct choice. For some reason (maybe historical), primitive types are not treated like any other type. In VSCode, they're actually under `storage` rather than `type`. I think the more correct change would be to modify the tree-sitter queries to highlight them under that scope.

**Comment by archseer** [comment]
> In VSCode, they're actually under `storage` rather than `type`. I think the more correct change would be to modify the tree-sitter queries to highlight them under an equivalent scope.

From what I remember I deliberately omitted `storage` as a toplevel member. There's `keyword.storage.type` for modifiers but built in types should just be `type.builtin`.

Highlighting doesn't need to be identical across editors, and given we're using a different engine (tree-sitter vs regex-based textmate grammar) it's unlikely we can get them 1:1 anyway.

**Comment by saccarosium** [comment]
For recap. Do I revert the built-in types to green?

**Comment by David-Else** [comment]
Is it possible to add the highlights for debugging that were introduced in https://github.com/helix-editor/helix/pull/5957? 
```
| Key                               | Notes                                                                                          |
| ---                               | ---                                                                                            |
| `ui.debug.breakpoint`             | Breakpoint indicator, found in the gutter                                                      |
| `ui.debug.active`                 | Indicator for the line at which debugging execution is paused at, found in the gutter          |
| `ui.highlight.frameline`          | Line at which debugging execution is paused at                                                 |

```

I am not sure exactly what they are, I looked at https://github.com/microsoft/vscode/blob/main/extensions/theme-defaults/themes/dark_plus.json. 

image

**Comment by kirawi** [comment]
Hey, sorry for the delay. I'll be looking at this within a few days.

**Comment by kirawi** [comment]
@David-Else I haven't used dark plus in a year or so. I'll be removing myself from maintainer of the theme after this PR. I think that your suggestions thus far have been consistent with VSCode, so if you're interested in becoming a new maintainer then I'd be fine with that.

**Comment by saccarosium** [comment]
At this point, I think we can close this. I don't really use helix anymore.

**Pull request closed without merging**